src/geo/projection/globe.js

```diff
@@ -104,36 +108,33 @@ class GlobeTileTransform {
     }
 
     upVector(id: CanonicalTileID, x: number, y: number): vec3 {
-        return new GlobeTile(id).upVector(x / EXTENT, y / EXTENT);
-    }
+        const corners = tileLatLngCorners(id);
+        const tl = corners[0];
+        const br = corners[1];
 
-    upVectorScale(id: CanonicalTileID): number {
-        const pixelsPerMeterECEF = mercatorZfromAltitude(1, 0.0) * 2.0 * globeRefRadius * Math.PI;
-        const maxTileScale = tileNormalizationScale(id);
-        return pixelsPerMeterECEF * maxTileScale;
-    }
+        const tlUp = latLngToECEF(tl[0], tl[1]);
+        const trUp = latLngToECEF(tl[0], br[1]);
+        const brUp = latLngToECEF(br[0], br[1]);
+        const blUp = latLngToECEF(br[0], tl[1]);
 
-    tileSpaceUpVectorScale(): number {
-        return mercatorZfromAltitude(1, this._tr.center.lat) * this._tr.worldSize;
-    }
+        vec3.normalize(tlUp, tlUp);
+        vec3.normalize(trUp, trUp);
+        vec3.normalize(brUp, brUp);
+        vec3.normalize(blUp, blUp);
 
-    _calculateGlobeMatrix() {
-        const localRadius = EXTENT / (2.0 * Math.PI);
-        const wsRadius = this._worldSize / (2.0 * Math.PI);
-        const s = wsRadius / localRadius;
+        const u = x / EXTENT;
+        const v = y / EXTENT;
 
-        const lat = clamp(this._tr.center.lat, -this._tr.maxValidLatitude, this._tr.maxValidLatitude);
-        const x = mercatorXfromLng(this._tr.center.lng) * this._worldSize;
-        const y = mercatorYfromLat(lat) * this._worldSize;
+        const tltr = vec3.lerp([], tlUp, trUp, u);
+        const blbr = vec3.lerp([], blUp, brUp, u);
```

Comment:
This will actually produce normal vectors that are slightly off because the relation between latitude and mercator coordinates is not linear. This has been fixed in gl-native to something like this:
```
vec3 GlobeProjection::elevationVector(double x, double y, const CanonicalTileID& id) const {
    auto tiles = 1 << id.z;
    auto mercX = static_cast<double>(x / util::EXTENT + id.x) / tiles;
    auto mercY = static_cast<double>(y / util::EXTENT + id.y) / tiles;

    return latLngToECEF(Projection::latFromMercatorY(mercY), Projection::lngFromMercatorX(mercX), 1.0);
}
```
This doesn't have to be fixed as part of this PR. Just wanted to bring this up so we'll remember to fix it. :)

Reply:
I'll give a look at fixing and testing this after the rebase.